src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AddAzureRmSqlManagedInstanceTransparentDataEncryptionCertificate.cs

```diff
@@ -0,0 +1,138 @@
+
+        /// <summary>
+        /// Intializes the model adapter
+        /// </summary>
+        /// <param name="subscription">The subscription the cmdlets are operation under</param>
+        /// <returns>The server adapter</returns>
+        protected override AzureSqlDatabaseTransparentDataEncryptionArmAdapter InitModelAdapter(IAzureSubscription subscription)
+        {
+            return new AzureSqlDatabaseTransparentDataEncryptionArmAdapter(DefaultProfile.DefaultContext);
+        }
+        
+        /// <summary>
+        /// Get entity returns null since the certificate has not been added.
+        /// </summary>
+        /// <returns>null, since the certificate does not exist</returns>
+        protected override IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> GetEntity()
+        {
+            return null;
+        }
+
+        /// <summary>
+        /// Constructs the model to send to the update API
+        /// </summary>
+        /// <param name="model"> Model to send to the update API</param>
+        /// <returns>The model to send to the update</returns>
+        protected override IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> ApplyUserInputToModel(IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> model)
+        {
+            // Construct a new entity so we only send the relevant data to the server
+            List<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> newEntity = new List<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel>();
+            newEntity.Add(new Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel()
+            {
+                ResourceGroupName = this.ResourceGroupName,
+                ManagedInstanceName = this.ManagedInstanceName,
+                PrivateBlob = this.PrivateBlob,
+                Password = this.Password
+            });
+            return newEntity;
+        }
+
+        /// <summary>
+        /// Sends the Firewall Rule update request to the service
+        /// </summary>
+        /// <param name="entity">The update parameters</param>
+        /// <returns>The response object from the service</returns>
+        protected override IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> PersistChanges(IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionCertificateModel> entity)
+        {
+            ModelAdapter.AddAzureRmSqlManagedInstanceTransparentDataEncryptionCertificate(entity.First());
+            return entity;
+        }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected override bool WriteResult() { return false; }
```

Comment:
@nivimsft is this what's written to the output stream no matter what? Where does this get called? We need to make sure that we write `true` to the output stream in the case of a successful cmdlet execution

Reply:
The return value of WriteResult determines if the object returned by persist changes. This is called in the AzureSqlCmdletBase class.